README overhaul

drakeltheryuujin opened the pull request.

drakeltheryuujin committed:
README overhaul

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+## Problem Statement
 
-You can see the commit we made was made in the context of the `new-feature` branch.
+A key to collaborating with git is to keep discrete and individual lines of work
```
ensure you're `ticking` `git`. There are multiple instances of this.

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+Best practices suggest that any new set of changes related to fixing a bug, creating a feature,
+or even messing around with experimental code in a "sandbox", should be started on a new branch.
+In order to start a new branch, in the terminal type `git branch <branch name>` to create the
+newly defined branch. In the case of a branch relating to isolating a new feature, you could name
```
isolating -> writing

seems simpler

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+or even messing around with experimental code in a "sandbox", should be started on a new branch.
+In order to start a new branch, in the terminal type `git branch <branch name>` to create the
+newly defined branch. In the case of a branch relating to isolating a new feature, you could name
+the branch `isolating-new-feature`. The master timeline remains unchanged and clean. This creates
```
`add-rainbow-background` -- something less abstract; make change throughout doc.

sgharms replied:
first use of timeline =~ branch here.  more consistent to stick with branch?

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
-Your local branches can attach to remote branches that live on the internet, generally on GitHub, that your team members might contribute to and you can download locally.
+In order to start making changes on your new branch, you need to "checkout" or move into the
+`isolating-new-feature` timeline or branch, so that git knows that all commits made apply to
+only that unit of work, timeline, or branch. You can move between branches with
```
same as above.

sgharms commented on README.md:
````diff
@@ -1,369 +1,137 @@
- * [new branch]      remote-feature-branch -> origin/remote-feature-branch
-```
+You can always move between branches with `git checkout <branch name>`. If you are currently on
+`isolating-new-feature`, you can move back to master with `git checkout master`. If the last
````
Cut from beginning to first comma.

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+it again using `git checkout isolating-new-feature`.
 
-From within `master` (though technically what branch I was in when I typed `git fetch -a` does not matter), I executed `git fetch -a`. The last 3 lines of output are really important, let's take a closer look:
+If you have changes that are not yet committed, when you switch between branches, the untracked
```
This is getting git-confusing here. Let's change this to be

"Be sure to make sure that all of your works in progress and new files are committed before your change branches. If you created scratch files you don't want, delete them. If you made changes that you're attached to, make sure to commit them. (new para.) 

The master branch..."

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+If you have changes that are not yet committed, when you switch between branches, the untracked
+changes will follow you to the next branch. These changes must be committed with `git commit -m "<my commit message>"`
+while still on the feature or bug branch the changes belong to. When the changes on your feature
+or bug branch are committed, you'll notice that when you are on master or another branch with a
```
do a global search and replace to verify you tick `master` consistently.

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+or bug branch are committed, you'll notice that when you are on master or another branch with a
+different timeline, the commit with those changes will not be present. The master branch only has
+the code from the most recent commit relative to the master timeline or branch. The code from our
+`isolating-new-feature` is tucked away in that branch, waiting patiently in isolation from the
```
"in _that_ branch"

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+different timeline, the commit with those changes will not be present. The master branch only has
+the code from the most recent commit relative to the master timeline or branch. The code from our
+`isolating-new-feature` is tucked away in that branch, waiting patiently in isolation from the
+rest of your code in `master` until the feature is considered complete.
```
(or any _other_ branch for that matter)

sgharms commented on README.md:
````diff
@@ -1,369 +1,137 @@
-   bfe50fc..0ae1da2  master     -> origin/master
- * [new branch]      remote-feature-branch -> origin/remote-feature-branch
-```
+The final step of completing the `isolating-new-feature` work sprint is to merging that timeline
````
haven't used sprint before, cut it.

sgharms commented on README.md:
````diff
@@ -1,369 +1,137 @@
+`isolating-new-feature` is tucked away in that branch, waiting patiently in isolation from the
+rest of your code in `master` until the feature is considered complete.
 
-```
````
"is to merge"
timeline -> branch

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+into the master timeline.
 
-The first line, `From github.com:aviflombaum/mission-critical-application` is informing us which remote our `git fetch` updated from, namely, the remote repository located at: https://github.com/aviflombaum/mission-critical-application
+## Explain Merging branches
```
Cap B-branches

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
 
-When we `fetch` with git, we are asking to copy all changes on the remote to our local git repository, but not actually integrate any. The next line, `bfe50fc..0ae1da2  master     -> origin/master` is telling us that a new commit was found in `origin/master`. `origin/master` means the GitHub version of `master`. Even though git fetched a new commit from `origin/master`, it did not merge it into t ...
+Now that you have some additions to the code that you'd like to combine back with the master set of
+code, the goal is to bring the timeline of commits that occurred on the `isolating-new-feature`
```
OK, i'm kinda OK with timeline now, you use it a lot. You could ignore other comments as long as you bring it out as a synonym to branch earlier and explicitly. I'd still suggest cutting it and keeping with git's terminology tho.

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+`master` because that is where you want the commits to end up.
 
-Our remote copy on GitHub has a file, `remote-bug-fix`, presumably some code that another developer pushed up to our remote version of the `master` branch to fix a bug. Even after we fetched, our local copy still doesn't appear to have that file.
+When performing `git merge`, `git` will ask you to create a commit to reflect that you've done
```
Let's not offer them this path. Direct them to `git merge -m "merge in feature add rainbow-background`"

so basically cut the hypothetical.

sgharms commented on README.md:
```diff
@@ -1,369 +1,137 @@
+a merge. If you use `git merge -m "<commit message>"` you can complete the commit in one action.
 
-After you fetch, you have access to the remote code but you still have to merge it. How do you merge a change fetched from `origin/master` into your current master? From within your local master branch, type: `git merge origin/master`, referring to the branch's full path, `remote/branch`, or `origin/master`.
+Now the branches have been merged. If you type `git log`, you'll see the commies from the
```
Commies? 🇷🇺 🇨🇳 

Also, spelling: commies -> commits ;)

sgharms commented:
@drakeltheryuujin Reviewed content. Definitely improves the content, but make sure you're consistent with applying styling, formatting, and using technical words you introduce consistently.

@jenmyers I'm not familiar with the overall arc at the moment, but I *think*  i might recommend to cut the `fetch` and `pull` parts of this doc for the FEWDS audience. The details are probably beyond their scope. It MIGHT be enough to say: 

1. You have a local git repository, but we've learned that the thing on github is just a remote too.
2. Consequently if a peer adds a new branch, you can get it and merge it into your branch. 
3. OR if a peer makes a great branch and merges it to master, you can update your master to get the latest changes.

At this point in your use of git, focus on using branches to contribute new features to your local master and push your changes to your remote master on GitHub. As you get more comfortable with git and its workflows, you can learn more about using git's advanced collaboration features.

drakeltheryuujin committed:
Ticks instances of git

drakeltheryuujin committed:
Additional changes from Steven feedback

sgharms committed:
Wip git collaboration updates sgh (#27)

* Formatting and typos

* Introduce "Golden Practices"

Focus specifically on branch-based development.
Walking away with practices around that > practices
around remote work.

* Add more "timeline" based fiction refs

* Clarify merging flow; technology

* Remove  pull / fetch section

It's really complex and goes over the head of the
audience at this level.

* Add simplified push / fetch preview

* Correct all-important gif link

* Wordsmith

* Introduce git log --graph earlier

Students had no means for checking history when
it's useful to see the before versus the after.

* Simplify branch switching

* Intro git log even earlier

* Add docs re: deletion b/c it appears in quiz

* Wc per Jen

sgharms commented:
This change is huge and different in purpose. It's a great document but cuts some information. Therefore, we're going to take this branch and fork it.
